config/VigraConfigureThreading.cmake

```diff
@@ -2,11 +2,20 @@ MACRO(VIGRA_CONFIGURE_THREADING)
     set(THREADING_FOUND 0)
     if(((APPLE AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS "4.7.0") OR MINGW)
          AND NOT WITH_BOOST_THREAD)
-        set(THREADING_FOUND 0)
-        if(${1} MATCHES "REQUIRED")
-            MESSAGE(FATAL_ERROR "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.")
-        else()
-            ADD_DEFINITIONS(-DVIGRA_SINGLE_THREADED)
+        TRY_COMPILE(THREADING_FOUND 
+            ${CMAKE_BINARY_DIR} ${PROJECT_SOURCE_DIR}/config/check_std_thread.cxx
+            OUTPUT_VARIABLE THREADING_TEST_OUTPUT )
+        if (NOT THREADING_FOUND)
+            if(${ARGN} MATCHES "REQUIRED")
+                MESSAGE(FATAL_ERROR "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.\n"
+                "For reference, the failed compiler output is shown below."
+                "${THREADING_TEST_OUTPUT}\n\n"
+                "Compiler does not support C++ threading. Try to run cmake with '-DWITH_BOOST_THREAD=1'.\n"
+                "For reference, the failed compiler output is shown above.\n"
+                )
```

Comment:
Ok. Ideally, I'd imagine output like this:
```
Checking for threading support:   std::thread 
     or
Checking for threading support:   boost::thread 
     or
Checking for threading support:   not found 
(compiling single threaded, consider running 'cmake -DWITH_BOOST_THREAD=1')
     or
Checking for threading support:   not found
***********************************************
   test program output
***********************************************
C++ threading is required, but your compiler doesn't support it.
For reference, the failed compiler output is shown above.
Try to run cmake with '-DWITH_BOOST_THREAD=1'.
```
It may be a bit of overkill, but you get the idea.